prov/cxi/src/cxip_dom.c

```diff
@@ -2005,6 +2008,23 @@ int cxip_domain(struct fid_fabric *fabric, struct fi_info *info,
 		cxi_domain->tclass = FI_TC_BEST_EFFORT;
 	}
 
+	/* Initialize CQ data sizes for messaging and RMA separately.
+	 * Both default to info->domain_attr->cq_data_size initially, but can be
+	 * controlled independently. This allows messaging to use remote CQ data
+	 * without forcing RMA writedata support.
+	 *
+	 * msg_cq_data_size: for FI_REMOTE_CQ_DATA in messaging operations
+	 * rma_cq_data_size: for fi_writedata/fi_inject_writedata operations
+	 */
+	cxi_domain->msg_cq_data_size = info->domain_attr->cq_data_size;
+	cxi_domain->rma_cq_data_size = cxip_env.enable_writedata ?
+		                       info->domain_attr->cq_data_size : 0;
```

Comment:
Since we require FI_MR_PROV_KEY to support RMA cq_data, we should just include that check here and indicate a warning if requested but FI_MR_PROV_KEY is not in use. This simplifies code else where. Something like this:
```
        if (cxip_env.enable_writedata && info->domain_attr->cq_data_size) {
                if (cxi_domain->util_domain.mr_mode & FI_MR_PROV_KEY)
                        cxi_domain->rma_cq_data_size =
                                info->domain_attr->cq_data_size;
                else
                        CXIP_WARN("fi_writedata support requires FI_MR_PROV_KEY\n");
        }
```
you could also move the setting of cxi_domain->is_prov_key above if you prefer.